integ/src/eks_provider.rs

```diff
@@ -124,9 +132,71 @@ pub async fn get_cluster_info(cluster_name: &str, region: &str) -> ProviderResul
         controlplane_sg,
         clustershared_sg,
         iam_instance_profile_arn,
+        dns_ip,
     })
 }
 
+async fn dns_ip(eks_client: &aws_sdk_eks::Client, cluster_name: &str) -> ProviderResult<String> {
+    let describe_results = eks_client
+        .describe_cluster()
+        .name(cluster_name)
+        .send()
+        .await
+        .context("Unable to get eks describe cluster")?;
+
+    let kubernetes_network_config = describe_results
+        .cluster
+        .as_ref()
+        .context("Response missing cluster field")?
+        .kubernetes_network_config
+        .as_ref()
+        .context("Cluster missing kubernetes_network_config field")?;
```

Comment:
non-blocking nit:
Whenever you run into situations where you want to retrieve deeply nested object, you can combine all the Option checks with [`and_then()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then).